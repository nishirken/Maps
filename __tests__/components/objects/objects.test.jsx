import { List } from 'immutable';
import { testInitialState } from 'Store';
import { Objects } from 'Components';
import StyledObjectNameField from 'Components/objects/styled-object-name-field';
import StyledCreateObjectButton from 'Components/objects/styled-create-object-button';

describe('Objects component', () => {
    const objects = List([
        testInitialState.get('getObjects').get(1).get('object'),
        testInitialState.get('getObjects').get(2).get('object'),
    ]);
    let TestObjects = {};

    beforeEach(() => {
        jest.resetModules();
        TestObjects = shallow(
            <Objects
                markerIndex={1}
                markerObjects={objects}
                objectDeleteIndexes={List([])}
            />
        );
    });

    it('Rendered', () => {
        expect(TestObjects).toMatchSnapshot();
    });

    it('Should rendered objects item', () => {
        expect(TestObjects.find('ObjectsItem').length).toBe(2);
    });

    it('Should filtered objects item by delete indexes', () => {
        TestObjects.setProps({ objectDeleteIndexes: List([1]) });
        expect(TestObjects.find('ObjectsItem').length).toBe(1);
    });

    it('Should render name field, when state has a createObject property', () => {
        TestObjects.setState({ createObject: true });
        expect(TestObjects.instance().renderCreateObject().type).toBe(StyledObjectNameField);
    });

    it('Should render create object button, when state doesn\'t has a createObject property', () => {
        TestObjects.setState({ createObject: false });
        expect(TestObjects.instance().renderCreateObject().type).toBe(StyledCreateObjectButton);
    });

    test('toggleCreateObject method', () => {
        TestObjects.setState({ createObject: false });
        TestObjects.instance().toggleCreateObject();
        expect(TestObjects.state('createObject')).toBeTruthy();
    });

    it('Should save object name value', () => {
        const event = {
            target: {
                value: 'object',
            },
        };

        TestObjects.instance().onChangeHandler(event);
        expect(TestObjects.instance().objectName).toBe(event.target.value);
    });

    it('Should validate empty entered object name', () => {
        expect(TestObjects.instance().objectNameValidate('newObject')).toBe('newObject');
        expect(TestObjects.instance().objectNameValidate('')).toBe('object');
    });

    it('Should toggle create object condition, when pushing esc or enter button', () => {
        const event = keyCode => {
            return {
                keyCode,
            };
        };

        TestObjects.setProps({ setObject: jest.fn() });
        TestObjects.instance().toggleCreateObject = jest.fn();
        TestObjects.instance().onKeyDownHandler(event(13));
        expect(TestObjects.instance().toggleCreateObject).toHaveBeenCalledTimes(1);
        TestObjects.instance().onKeyDownHandler(event(27));
        expect(TestObjects.instance().toggleCreateObject).toHaveBeenCalledTimes(2);
    });

    it('Should focused on the input, when update', () => {
        TestObjects.setState({ createObject: true });
        TestObjects.instance().input = {
            focus: jest.fn(),
        };

        TestObjects.instance().componentDidUpdate();
        expect(TestObjects.instance().input.focus).toHaveBeenCalled();
    });
});
