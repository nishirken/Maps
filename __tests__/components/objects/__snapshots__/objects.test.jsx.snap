// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Objects component Rendered 1`] = `
<styled.div>
  <styled.div
    onClick={[Function]}
  >
    create new object
  </styled.div>
  <ObjectsItem
    index={0}
    markerIndex={1}
    name="Object for 1"
    number={1}
  />
  <ObjectsItem
    index={1}
    markerIndex={1}
    name="Object for 1"
    number={2}
  />
</styled.div>
`;
